src/controllers/request-breach-scan.js

```diff
@@ -0,0 +1,72 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { getMessage } from '../utils/fluent.js'
+import { getBreachLogo } from '../utils/breach-logo.js'
+import { UserInputError } from '../utils/error.js'
+import { getSha1 } from '../utils/fxa.js'
+import { getBreachesForEmail } from '../utils/hibp.js'
+
+/**
+ * @typedef {{ success: false }} RequestBreachScanErrorResponse
+ * @typedef {{ success: true, breaches: Array<unknown>, heading: string, logos: string[], dataClassStrings: string[][], total: number }} RequestBreachScanSuccessResponse
+ * @typedef {RequestBreachScanErrorResponse | RequestBreachScanSuccessResponse} RequestBreachScanResponse
+ */
+
+/** @type {import('express').RequestHandler<unknown, RequestBreachScanResponse>} */
+async function requestBreachScan (req, res, next) {
+  if (req.method !== 'POST' || typeof req.body?.email !== 'string') {
+    return next()
+  }
+
+  // TODO could share this validation logic with add-email API?
+  const email = req.body.email
+  // Use the same regex as HTML5 email input type
+  // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email#basic_validation
+  const emailRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/
```

Comment:
I think this is a good common-sense check, but I don't think it's sanitized enough to make it safe to append to the DOM as a string.

We might want to try to not duplicate it and have a utility function also.

Reply:
Dammit this was your code :P But yes, moved this into a separate validation function and reused it in both places: e00e5079c

As for not using it to inject into the DOM, see https://github.com/mozilla/blurts-server/pull/2968#discussion_r1159668495.